File: app/src/main/java/app/endershrooms/thatcatapp/db/dao/CatDao.java
package app.endershrooms.thatcatapp.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import app.endershrooms.thatcatapp.model.Cat;
import java.util.List;

@Dao
public interface CatDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  void insert(Cat... cats);

  @Query("SELECT * FROM Cat")
  LiveData<List<Cat>> getAllCats();

}

File: app/src/main/java/app/endershrooms/thatcatapp/db/dao/CatDao.kt
package app.endershrooms.thatcatapp.db.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import app.endershrooms.thatcatapp.model.Cat

@Dao
interface CatDao {

    @Query("SELECT * FROM Cat")
    fun getAllCats(): LiveData<List<Cat>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg cats: Cat)

}
